rakelib/docker_based_test.rake

```
@@ -0,0 +1,107 @@
+      ENV["GOO_PORT"]="10035"
+      ENV["GOO_PATH_QUERY"]="/repositories/ontoportal_test"
+      ENV["GOO_PATH_DATA"]="/repositories/ontoportal_test/statements"
+      ENV["GOO_PATH_UPDATE"]="/repositories/ontoportal_test/statements"
+      ENV["COMPOSE_PROFILES"]="ag"
+      Rake::Task["test:docker:up"].invoke
+      # AG takes some time to start and create databases/accounts
+      # TODO: replace system curl command with native ruby code
+      unless system("curl -sf http://127.0.0.1:10035/repositories/ontoportal_test/status | grep -iqE '(^running|^lingering)' || exit 1")
+        printf("waiting for AllegroGraph container to initialize")
+        sec = 0
+        until system("curl -sf http://127.0.0.1:10035/repositories/ontoportal_test/status | grep -iqE '(^running|^lingering)' || exit 1") do
```

RuboCop (1.73)

[convention] Do not use `do` with multi-line `until`.

(Style/WhileUntilDo)